src/System.Private.CoreLib/src/System/String.Searching.cs

```diff
@@ -210,24 +224,33 @@ private static unsafe void InitializeProbabilisticMap(uint* charMap, char[] anyO
         {
             bool hasAscii = false;
 
+            charMap[0] = 0;
```

Comment:
Is there and 64 but #define?

Could cast to ulong* and init 4?

Reply:
For even better performance I would suggest using a dedicated struct (`[StructLayout(LayoutKind.Explicit, Size = 32)] private struct ProbabilisticMap { }`) - then zero-init would be `*charMap = default` and stackallock overhead (stack canaries) would be eliminated also.

Reply:
I notice that `Block16` and `Block64` is disabled in `Buffer` for ARM. I'm not sure of the reason for that. Anything to be concerned with?

Reply:
https://github.com/dotnet/coreclr/pull/13793 enables HAS_CUSTOM_BLOCKS for ARM64, maybe @sdmaclea knows what's the catch on ARM32? Maybe it's slow for copying, but zero-init is still fast?

Reply:
> maybe @sdmaclea knows what's the catch on ARM32?

There is no catch.  It should be put on the arm32 todo list.  I do not have an arm32 platform so I do I do not presume to optimize it.  I think block unroll was recently enabled for arm32.  I suspect the optimization should now be useful for arm32, but someone would need to test...